pkg/validation/internal/object.go

```diff
@@ -0,0 +1,199 @@
+		}
+	}
+
+	minAvailable := pdb.Spec.MinAvailable
+	if minAvailable != nil {
+		if minAvailable.StrVal == "100%" {
+			result.Add(errors.ErrInvalidObject(pdb, "minAvailable field cannot be set to 100%"))
+		}
+	}
+
+	return
+}
+
+// validatePriorityClass checks the PriorityClass object to ensure globalDefault is set to false.
+// See https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/adding-priority-classes.md
+func validatePriorityClass(u *unstructured.Unstructured) (result errors.ManifestResult) {
+	pc := schedulingv1.PriorityClass{}
+
+	b, err := u.MarshalJSON()
+	if err != nil {
+		result.Add(errors.ErrInvalidParse("error converting unstructured", err))
+		return
+	}
+
+	err = json.Unmarshal(b, &pc)
+	if err != nil {
+		result.Add(errors.ErrInvalidParse("error unmarshaling priorityclass", err))
+		return
+	}
+
+	if pc.GlobalDefault {
+		result.Add(errors.ErrInvalidObject(pc, "globalDefault field cannot be set to true"))
+	}
+
+	return
+}
+
+func validateRBAC(u *unstructured.Unstructured) (result errors.ManifestResult) {
+	var policyRules []rbacv1.PolicyRule
+
+	b, err := u.MarshalJSON()
+	if err != nil {
+		result.Add(errors.ErrInvalidParse("error converting unstructured", err))
+		return
+	}
+
+	switch u.GroupVersionKind().Kind {
+	case RoleKind:
+		role := rbacv1.Role{}
+		err = json.Unmarshal(b, &role)
+		if err != nil {
+			result.Add(errors.ErrInvalidParse("error unmarshaling role", err))
+			return
+		}
+		policyRules = role.Rules
+	case ClusterRoleKind:
+		clusterrole := rbacv1.ClusterRole{}
+		err = json.Unmarshal(b, &clusterrole)
+		if err != nil {
+			result.Add(errors.ErrInvalidParse("error unmarshaling clusterrole", err))
+			return
+		}
+		policyRules = clusterrole.Rules
+	}
+
+	return audit(policyRules)
+}
+
+// audit checks the provided rbac policies against prescribed limitations.
+// If permission is granted to create/modify a PDB, a warning is returned.
+// If permission is granted to modify default SCCs in OpenShift, an error is returned.
+func audit(policies []rbacv1.PolicyRule) (result errors.ManifestResult) {
+	// check for permission to modify/create PDBs
+	for _, rule := range policies {
+		if contains(rule.APIGroups, PodDisruptionBudgetAPIGroup) &&
+			contains(rule.Resources, "poddisruptionbudgets") &&
+			contains(rule.Verbs, rbacv1.VerbAll, "create", "update", "patch") {
+			result.Add(errors.WarnInvalidObject("RBAC includes permission to modify/create poddisruptionbudgets", rule))
+		}
+	}
+
+	// check sccs for modifying default known SCCs
+	for _, rule := range policies {
+		if contains(rule.APIGroups, SCCAPIGroup) &&
+			contains(rule.Resources, "securitycontextconstraints") &&
+			contains(rule.Verbs, rbacv1.VerbAll, "delete", "update", "patch") &&
+			containsDefaults(rule.ResourceNames, defaultSCCs) {
+			result.Add(errors.ErrInvalidObject(rule, "RBAC includes permission to modify default SCCs"))
```

Comment:
Spell out uncommon initialisms
```suggestion
			result.Add(errors.ErrInvalidObject(rule, "RBAC includes permission to modify default securitycontextconstraints"))
```